### DISCUSSION TITLE
Problems with plugin translation

### DISCUSSION THREAD

**Discussion by willinrecovery** [code4recovery/12-step-meeting-list#1394] [Q&A]
We are in Canada and so some of our users speak French as their first language. Our whole site is not bilingual, but because of this plugin it seems our meetings page is!  So first, yay! I didn't expect this.

However, there were a couple of problems.

First, one of the meeting types seems to be substituted with another in French. This is a Debtors Anonymous site but, in French, the Business Owner meetings are showing as Big Book meetings — but only in French. Is this a bug? Or do I just need to configure something else to make it work correctly in French. We have International members as well, should I configure multiple languages?

Also, there is trouble with the meeting type descriptions. I had substituted new descriptions for Closed meeting to reflect our program, but Closed meetings for example seem to have a French version of the original AA description. I am guessing there is more for me to configure here. Once again, how many languages should I configure — and how?

Thank you so much for what you do and all your patience!

**Comment by SIA-WEBSUP** [comment]
"Business Owner" being translated as "Gros Livre" is likely a bug. Thank you for bringing this to our attention.

I'm not sure what you mean by "I had substituted new description for Closed meeting...". 
Where exactly did you make this substitution?

**Reply by SIA-WEBSUP** [reply]
Sorry, I think I see the description that you 'customized', but I was unaware that the description could be customized. Where did you enter this text? Is it in the theme's function.php?

**Reply by willinrecovery** [reply]
Yes, that is where I made the changes (in functions.php). I do not recall where I found the direction for how to do this.

<img width="672" alt="Monosnap CECDA Staging 2024-03-21 12-10-27" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/636277e7-8d65-4724-8ee9-936878877ee8">

<img width="1120" alt="Monosnap Monosnap 2024-03-21 12-16-45" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/57a6c299-bde1-4c67-9a46-c9f569a0a527">

**Reply by willinrecovery** [reply]
"Business owner" does seem to be a version of the "Big Book" AA meeting type — at least in the sense that they both use 'B'.

 I gather the French version is still drawing from AA categories. Honestly, until a member noticed this, I didn't realize the plugin supported multiple browser languages.

While this bug is being worked out, do you think it would be possible for me to customize the French type and description for 'B' to be a French version of "Business owner" for the AA category? I would have to do that for the DA meeting type in French anyway but I don't know how to do that.

As I am sure you can understand, the last thing I want to do is offend our French members.

**Comment by SIA-WEBSUP** [comment]
I'm still figuring out how WP handles translations, but what I can tell you is that the plugin doesn't house its own set of translation strings. It relies on WP to do the translation and the process to add a new string to the 'translation library' for a given plugin is not as straightforward as one might imagine.

I was thinking a workaround that would be rather straightforward would be to add a custom type for 'Business Owner' that has the type in English and French. Unfortunately this wouldn't be a great solution for the description unless you were willing to have the description in English and French which I imagine would be a little long. 

Here is the code that I was imagining:
```
if (function_exists('tsml_custom_types')) {
	tsml_custom_types(array(
		'BOA' => "Business Owner / Propriétaire d'entreprise",
	));
}
```

**Reply by SIA-WEBSUP** [reply]
Now I'm not 100% if TSML UI supports the custom types, but give it a try.

**Reply by willinrecovery** [reply]
Hmm. I can try that, but, presuming I understand what you have said and presuming I don't re-use a type code which AA uses, wouldn't we expect that the same translation process would pick up whatever description I provide and translate it?

If the plugin is not providing translation strings, the translation process is picking up strings from the AA types rather than the DA types.  If that is right, I think the best I could hope for would be that if a type code is not reused (like 'B' for Big Book and for Business owner) it will pick up the custom type description I have configured for this site and translate it into French (rather than picking up the AA name and description which wouldn't exist).

Does that make sense?

If it does, is it possible to "turn off" the existing B / Business owner type so we don't get two "Business owner" types in the backend?

**Reply by SIA-WEBSUP** [reply]
What I was proposing was that you use the custom type INSTEAD of the 'B' type and that the string holding the 'name' for this custom type would not need a translation because it would have both languages.

But you're right that there is likely a better solution!

**Reply by willinrecovery** [reply]
Just to clarify, if I do create a custom code to replace the 'B' type, is there a way to turn off the 'B' type?  Otherwise I will have two "Business owner" types in English. That will be confusing for future admins.

**Reply by SIA-WEBSUP** [reply]
Ty unseting 'B'. This may remove the option to set this type in the admin screen. 

```
if (function_exists('tsml_custom_types')) {
	tsml_custom_types(array(
	       'B' => '',
		'BOA' => "Business Owner / Propriétaire d'entreprise",
	));
}
```

**Reply by SIA-WEBSUP** [reply]
I just tested this and it didn't work as I expected. The checkbox is still there. 
I suppose you could use the following instead... but  this workaround is looking kludgier and kludgier.
```
		'B' => '~DO NOT USE ME~',
```

**Reply by willinrecovery** [reply]
Could you please remind me of where to find the documentation on how to customize the types? I promise I will bookmark it. Clearly I found it once!

Also, I noticed another plugin 12 Step Meeting List Xtras. Is this a possible solution? It seems to be add new codes and to turn off others.

Or would you suggest I keep doing this with code?

Thank you so much

**Reply by SIA-WEBSUP** [reply]
readme.txt
 - https://github.com/code4recovery/12-step-meeting-list/blob/main/readme.txt
    
which is also available on the wordpress.org page for the plugin
 -  https://wordpress.org/plugins/12-step-meeting-list/#my%20meeting%20type%20isn%27t%20listed%21
   
I wish I knew more about Xtras plugin. Sorry I can't help you there.

**Reply by willinrecovery** [reply]
Do you know if this issue has been flagged as a bug then?

Thank you for directing me to that documentation, it helped.  And this approach, as kludgy as it may be, seems to have worked.  

I changed the name of 'B' to 'Not in use'

I created a new type 'BDA' with its own description just in English.

I am not that familiar with the language settings in Chrome (there are a lot of options) but from my own testing, the browser now appears to be translating the new description rather than the default AA description (presumably because 'BDA' doesn't have an AA description). I am going to ask a francophone to test to see if they get the same results as their browser may have slightly different settings.

I am contemplating leaving the settings to AA or creating my own English and French meeting types for D.A. I am a little concerned about future web admins not knowing how to configure a French meeting.

<img width="1038" alt="Monosnap CECDA Staging 2024-03-22 19-04-14" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/5954d2dd-31e1-4daf-ba58-bed7b5d91aee">

<img width="1004" alt="Monosnap CECDA Staging 2024-03-22 19-25-42" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/de258e41-8e75-484a-9375-a019f58cb187">

<img width="987" alt="Monosnap CECDA Staging 2024-03-22 19-23-13" src="https://github.com/code4recovery/12-step-meeting-list/assets/159723102/976f804c-478c-4764-8d4c-77bbeae8f7e7">

**Reply by SIA-WEBSUP** [reply]
Terrific that the custom type names and custom descriptions are being translated!
If a bug report has *not* been opened up on this issue yet, one will be shortly. Glad you found a satisfactory workaround for now!

**Reply by SIA-WEBSUP** [reply]
Here's that issue in the TSML-UI repo: https://github.com/code4recovery/tsml-ui/issues/371
